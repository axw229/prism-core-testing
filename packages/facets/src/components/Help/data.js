// @flow
import { shouldAllowFeature } from '../../shared/utils/featureSwitch.util'
import { FEATURE_EXCLUSIONS } from '../../constants/configurations'

export const helpTabs = [{
  'id': 0,
  'name': 'ICONS_&_BUTTONS',
  'header': 'HELPFUL_HINTS.HEADER.ICONS_&_BUTTONS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ICONS_&_BUTTONS',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'plus-circle', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.ADD_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.ADD_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'folder', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.SAVE']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'trash', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.TRASH_A_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.TRASH_A_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'clone', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.MORE_SCENES',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.MORE_SCENES']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'align-justify', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.COLOR_DETAILS',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.COLOR_DETAILS']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'signal-3', 'rotate': 235, 'size': 'lg' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.GRAB_&_REORDER',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.GRAB_&_REORDER']
  }, {
    'fontAwesomeIcon': {},
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.PAINT_SCENE',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 1,
  'name': 'MY_COLOR_PALETTE',
  'header': 'HELPFUL_HINTS.HEADER.MY_COLOR_PALETTE',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.MY_COLOR_PALETTE',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'myColorPaletteBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPalette3'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.TAP_AND_HOLD_THE_BOTTOM_EDGE_OF_A_COLOR_SWATCH_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette4'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette5'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.MY_COLOR_PALETTE',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.MY_COLOR_PALETTE'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 2,
  'name': 'ADDING_COLORS',
  'header': 'HELPFUL_HINTS.HEADER.ADDING_COLORS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ADDING_COLORS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'addColorBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColor1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColor2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColor3'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.ADDING_COLORS',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.ADDING_COLORS'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColorMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 3,
  'name': 'COLOR_DETAILS',
  'header': 'HELPFUL_HINTS.HEADER.COLOR_DETAILS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.COLOR_DETAILS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'colorDetailBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetail1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetail2'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile2'
  }],
  'isHiddenMobile': false
}, {
  'id': 4,
  'name': 'PAINTING_MY_OWN_PHOTO',
  'header': 'HELPFUL_HINTS.HEADER.PAINTING_MY_OWN_PHOTO',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.PAINTING_MY_OWN_PHOTO',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'fill-drip', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINT_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINT_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'brush', 'rotate': 45 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINTBRUSH',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINTBRUSH']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'mouse-pointer', 'rotate': -20, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.SELECT',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.SELECT']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'eraser', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ERASE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ERASE']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'plus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DEFINE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.DEFINE_AREA']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'minus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REMOVE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.REMOVE_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'search-plus', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ZOOM',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ZOOM']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'undo-alt', 'rotate': 0 }, {
      'variant': 'fa',
      'icon': 'redo-alt',
      'rotate': 0
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNDO_&_REDO',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.UNDO_&_REDO'],
    'isUndoRedo': true
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'redo-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REDO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'eye', 'flip': 'horizontal' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.HIDE_PAINT',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'info-circle' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.INFO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'trash-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DELETE_GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-group' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-ungroup' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNGROUP',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 5,
  'name': 'SAVING_MY_WORK',
  'header': 'HELPFUL_HINTS.HEADER.SAVING_MY_WORK',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.SAVING_MY_WORK',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'savingBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'saving1'
  }, { 'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO', 'role': 'text', 'imagePathKey': 'saving3' }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO',
    'role': 'text',
    'imagePathKey': 'savingMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'savingMobile2'
  }],
  'isHiddenMobile': false
}]

export const helpHeader = 'HELPFUL_HINTS.TITLE'

export const filterHelpItems = (contents: string[], contentsHiddenMobile: string[], featureExclusions: string[]): typeof helpTabs => {
  // Filtering Help Section details:
  // Icons & Buttons - Save hint
  !shouldAllowFeature(featureExclusions, FEATURE_EXCLUSIONS.documentSaving) && helpTabs[0].content.forEach((icon, index) => {
    icon.iconInfoName === 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE' && helpTabs[0].content.splice(index, 1)
  })

  const helpTabNames = helpTabs.map(item => item.name)
  contentsHiddenMobile.forEach(content => { helpTabs[helpTabNames.indexOf(content)].isHiddenMobile = true })

  return helpTabs.filter(item => contents.indexOf(item.name) !== -1)
}
