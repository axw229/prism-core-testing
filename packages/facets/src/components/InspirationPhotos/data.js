// @flow
import tinycolor from '@ctrl/tinycolor'
import { type Color } from 'src/shared/types/Colors.js.flow'
import uniqueId from 'lodash/uniqueId'

export const throttleDragTime = 5
// we need this two const to make sure cursor always point to the center of preview circle
export const activedPinsHalfWidth = 24

export const cloneColorPinsArr = (pins) => {
  return pins.map(pin => ({ ...pin }))
}

export const getRGBInitPins = (acgColors: Array<Object>) => {
  return acgColors.map<Object>((acgColor: any): Object => {
    return {
      rgbArray: [acgColor.r, acgColor.g, acgColor.b],
      rgbValue: `rgb(${acgColor.r},${acgColor.g},${acgColor.b})`,
      translateValueX: acgColor.x,
      translateValueY: acgColor.y
    }
  })
}

export const rgb2lab = (rgb: Array<any>) => {
  let r = rgb[0] / 255
  let g = rgb[1] / 255
  let b = rgb[2] / 255
  let x; let y; let z

  r = (r > 0.04045) ? Math.pow((r + 0.055) / 1.055, 2.4) : r / 12.92
  g = (g > 0.04045) ? Math.pow((g + 0.055) / 1.055, 2.4) : g / 12.92
  b = (b > 0.04045) ? Math.pow((b + 0.055) / 1.055, 2.4) : b / 12.92

  x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047
  y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000
  z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883

  x = (x > 0.008856) ? Math.pow(x, 1 / 3) : (7.787 * x) + 16 / 116
  y = (y > 0.008856) ? Math.pow(y, 1 / 3) : (7.787 * y) + 16 / 116
  z = (z > 0.008856) ? Math.pow(z, 1 / 3) : (7.787 * z) + 16 / 116

  return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)]
}

export type RenderingPin = {
  isActiveFlag: boolean,
  name: string,
  colorNumber: number,
  rgbValue: string,
  pinNumber: number,
  isContentLeft: boolean,
  translateX: number,
  translateValueY: number,
  pinId: string
}

export const renderingPins = (initPins: Array<Object>, canvasOffsetWidth: number, canvasOffsetHeight: number, colors: Color[]): RenderingPin[] => {
  return getRGBInitPins(initPins).map<Object>((acgColor: any, index: number): Object => {
    const color: Color = findClosestColor(acgColor.rgbArray, colors)
    const calculateTranslateX = acgColor.translateValueX * canvasOffsetWidth
    const calculateTranslateY = acgColor.translateValueY * canvasOffsetHeight

    return {
      ...color,
      isActiveFlag: (initPins.length - 1 === index),
      rgbValue: `rgb(${color.red},${color.green},${color.blue})`,
      pinNumber: index,
      pinId: uniqueId('pin_'),
      isContentLeft: calculateTranslateX < canvasOffsetWidth / 2,
      translateX: Math.min(canvasOffsetWidth - 2 * activedPinsHalfWidth, calculateTranslateX),
      translateY: Math.min(canvasOffsetHeight - 2 * activedPinsHalfWidth, calculateTranslateY)
    }
  })
}

export const findClosestColor = ([red, green, blue]: [number, number, number], colors: Color[]): Color => {
  const currentPixelHex: string = tinycolor(`rgb (${red}, ${green}, ${blue})`).toHex()

  return colors.reduce(([closestColor, smallestDistanceSoFar], color): [Color, number] => {
    const colorDistance: number = measureDistance(currentPixelHex, tinycolor(`rgb(${color.red},${color.green},${color.blue})`).toHex())
    return colorDistance < smallestDistanceSoFar ? [color, colorDistance] : [closestColor, smallestDistanceSoFar]
  }, [colors[0], 99999])[0]
}

export const measureDistance = (color1: string, color2: string): number => {
  const position1 = getChromaLuma(color1)
  const position2 = getChromaLuma(color2)
  const redDelta = position1[0] - position2[0]
  const greenDelta = position1[1] - position2[1]
  const lumaDelta = position1[2] - position2[2]
  const distance = redDelta * redDelta + greenDelta * greenDelta + lumaDelta * lumaDelta
  return distance
}

export const getChromaLuma = (rgbString: string) => {
  const color = tinycolor(rgbString)
  const luminosity = color.getLuminance()
  const colorRgb = color.toRgb()
  const red = colorRgb.r
  const green = colorRgb.g
  const blue = colorRgb.b
  const sum = red + green + blue
  let redChroma
  let greenChroma
  if (sum > 25) {
    redChroma = red / sum
    greenChroma = green / sum
  } else {
    redChroma = 1 / 3
    greenChroma = 1 / 3
  }

  return [redChroma, greenChroma, luminosity]
}
