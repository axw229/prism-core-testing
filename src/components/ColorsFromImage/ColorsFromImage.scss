.match-photo-picker {
  &__wrapper {
    width: 100%;
    /* @todo - This just stopped the annoy jank, needs to be properly fixed. -RS */
    cursor: pointer;
  }
  &__canvas {
    &--portrait {
      margin: 0 auto;
      text-align: center;
      display: block;
    }
  }
}