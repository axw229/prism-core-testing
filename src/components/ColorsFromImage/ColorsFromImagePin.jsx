// @flow
import React, { PureComponent, Fragment } from 'react'

type Props = {
  pinType: string,
  onClickMethod: Function,
  // activePinIndex: number,
  isActiveFlag: boolean,
  transformValue: string,
  RGBstring: string,
  previewColorName: string,
  previewColorNumber: string
}

class ColorsFromImagePin extends PureComponent<Props> {
  render () {
    // eslint-disable-next-line one-var
    const { previewColorName, previewColorNumber, transformValue, RGBstring, onClickMethod, isActiveFlag, pinType } = this.props

    let isPreviewPin = false
    let additionalPinClass = ''
    let isActiveClass = ''

    if (pinType === 'preview') {
      isPreviewPin = true
      additionalPinClass = 'pin--preview__wrapper'
    }

    if (isPreviewPin) {
      if (isActiveFlag) isActiveClass = 'pin--active__wrapper'
    } else {
      // TODO: fix this functionality
      // is `this.key` supposed to be a reference to the key prop? is it supposed to be internally tracked?
      // if (activePinIndex === this.key) isActiveClass = 'pin--active__wrapper'
    }

    return (
      <Fragment>
        <button onClick={onClickMethod} style={{ transform: transformValue }} className={additionalPinClass + ' pin__wrapper ' + isActiveClass}>
          <span className='pin__chip' style={{ background: RGBstring }} />
          <div className='pin__name-wrapper'>
            <span className='pin__copy pin__name'>{previewColorName}</span>
            <span className='pin__copy pin__number'>SW {previewColorNumber}</span>
          </div>
        </button>
      </Fragment>
    )
  }
}

export default ColorsFromImagePin
