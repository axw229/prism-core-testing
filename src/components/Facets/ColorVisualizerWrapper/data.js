export const renderingData = {
  color: {
    content: {
      title: 'Explore all the ways to start building My Color Palette.',
      subContent: [
        {
          image: '/src/images/color-visualizer-wrapper/color-submenu__thumbnail--explore-color.png',
          subTitle: 'DIGITAL COLOR WALL',
          subContent: 'Search Sherwin-Williams paint colors to add to your palette or get color info.',
          url: '/active/color-wall/section/sherwin-williams-colors'
        },
        {
          image: '/src/images/color-visualizer-wrapper/color-submenu__thumbnail--color-collections.png',
          subTitle: 'COLOR COLLECTIONS',
          subContent: 'Find colors that work together seamlessly throughout your home.',
          url: '/color-collections'
        },
        {
          image: '/src/images/color-visualizer-wrapper/color-submenu__thumbnail--match-photo.png',
          subTitle: 'MATCH A PHOTO',
          subContent: 'Capture colors from any photo and match them to real paint colors.',
          url: '/match-photo'
        }
      ]
    }
  },
  inspiration: {
    content: {
      title: 'Find inspiration for your next painting project.',
      subContent: [
        {
          image: '/src/images/color-visualizer-wrapper/inspiration-submenu__thumbnail--painted-scenes.png',
          subTitle: 'PAINTED PHOTOS',
          subContent: 'See colors in context to start building your color confidence.',
          url: '/color-from-image'
        },
        {
          image: '/src/images/color-visualizer-wrapper/inspiration-submenu__thumbnail--expert-color-picks.png',
          subTitle: 'EXPERT COLOR PICKS',
          subContent: 'Find expertly designed color palettes to use in your space.',
          url: '/expert-colors'
        },
        {
          image: '/src/images/color-visualizer-wrapper/inspiration-submenu__thumbnail--sample-photos.png',
          subTitle: 'INSPIRATIONAL PHOTOS',
          subContent: 'Use our images to discover colors from around the world.',
          url: '/color-from-image'
        }
      ]
    }
  },
  scenes: {
    content: {
      title: 'Start Painting.',
      subContent: [
        {
          image: '/src/images/color-visualizer-wrapper/scene-submenu__thumbnail--sample-scenes.png',
          subTitle: 'USE OUR PHOTOS',
          subContent: 'Virtually paint a sample photo.',
          url: '/paint-scene'
        },
        {
          image: '/src/images/color-visualizer-wrapper/scene-submenu__thumbnail--my-photos.png',
          subTitle: 'UPLOAD YOUR PHOTO',
          subContent: 'Virtually paint your own photo.',
          description: 'Photo Tips: For best results make sure your room is well lit. Natural light works best.',
          url: '/paint-scene'
        }
      ]
    }
  }
}
