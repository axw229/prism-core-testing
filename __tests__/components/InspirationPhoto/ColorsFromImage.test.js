import React from 'react'
import { shallow } from 'enzyme'
import ColorsFromImage from 'src/components/InspirationPhotos/ColorsFromImage'
import ColorsFromImagePin from 'src/components/InspirationPhotos/ColorsFromImagePin'

import { getImagesCollectionsData, collectionTabs } from 'src/components/InspirationPhotos/data'

const { collectionData } = getImagesCollectionsData(collectionTabs, 'tab1')
let defaultProps = {
  data: collectionData[0],
  isActivedPage: true
}

const createColorsFromImage = (props) => {
  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<ColorsFromImage {...newProps} />, { disableLifecycleMethods: true })
}

const pinnedColors = [
  { colorName: 'Rookwood Dark Red',
    colorNumber: '2801',
    isActiveFlag: true,
    isContentLeft: true,
    pinNumber: 0,
    rgbValue: 'rgb(75,41,41)',
    translateX: 35.74944320712695,
    translateY: 56.531827515400416 }]

const wrapper = createColorsFromImage()

describe('Testing state for createColorsFromImage component', () => {
  it('State should initialize correctly', () => {
    const initialState = wrapper.instance().state
    expect(initialState).toEqual({
      previewPinIsUpdating: false,
      previewPinIsActive: false,
      previewColorName: '',
      previewColorNumber: '',
      cursorX: 0,
      cursorY: 0,
      mappedCanvasIndex: 0,
      currentPixelRGB: [0, 0, 0],
      currentPixelRGBstring: 'rgb(0,0,0)',
      currentBrandColorIndex: 0,
      pinnedColors: [],
      imageStatus: 'loading',
      isDragging: false,
      position: { x: 0, y: 0, left: 0, right: 0, top: 0, bottom: 0 }
    })
  })
})

describe('compoments rendring test', () => {
  wrapper.instance().updateWindowDimensions = jest.fn()
  wrapper.instance().componentDidMount()
  it('should rendering ColorsFromImage Component', () => {
    expect(wrapper.find('.scene__image__wrapper').exists()).toBe(true)
  })

  it('should rendering Colors From Image pins', () => {
    wrapper.setState({
      isActivedPage: true,
      pinnedColors: pinnedColors
    })
    expect(wrapper.find(ColorsFromImagePin).exists()).toBe(true)
  })

  it('should rendering indicators', () => {
    wrapper.setState({
      isActivedPage: true,
      isDragging: true
    })
    expect(wrapper.find('.scene__image__wrapper__indicator').exists()).toBe(true)
  })

  it('should rendering delete button', () => {
    expect(wrapper.find('.scene__image__wrapper__delete-pin').exists()).toBe(true)
  })
})

describe('component event test', () => {
  it('click canvas should works fine', () => {
    const click = jest.fn()
    wrapper.instance().handleClick = click
    wrapper.instance().forceUpdate()
    wrapper.find('.scene__image__wrapper').simulate('click')
    expect(click).toHaveBeenCalled()
  })

  it('draging color pins should be dragable', () => {
    wrapper.setState({
      isActivedPage: true,
      pinnedColors: pinnedColors
    })
    let canvasOffset = { x: 0, y: 0 }
    wrapper.instance().imageDataData = [0, 0, 0]
    window.sessionStorage.setItem('canvasOffset', JSON.stringify(canvasOffset))
    const drag = jest.fn()
    wrapper.instance().handleDrag = drag
    wrapper.instance().forceUpdate()
    wrapper.find(ColorsFromImagePin).props().handleDrag()
    expect(drag).toHaveBeenCalled()
  })

  it('color pins should be deletable', () => {
    wrapper.setState({
      isActivedPage: true,
      pinnedColors: pinnedColors
    })
    const deletePin = jest.fn()
    wrapper.instance().deleteCurrentPin = deletePin
    wrapper.instance().forceUpdate()
    wrapper.find(ColorsFromImagePin).props().deleteCurrentPin()
    expect(deletePin).toHaveBeenCalled()
  })

  it('color pins could be removed', () => {
    // eslint-disable-next-line no-undef
    const e = new Event('test')
    expect(wrapper.instance().state.pinnedColors).toHaveLength(1)
    wrapper.instance().pinRemove(e)
    const PinnedColorListLen = wrapper.instance().state.pinnedColors.length
    expect(PinnedColorListLen).toEqual(0)
  })

  it('color pins could be added', () => {
    const wrapper = createColorsFromImage()
    wrapper.setState({
      isActivedPage: true,
      pinnedColors: pinnedColors
    })
    expect(wrapper.instance().state.pinnedColors).toHaveLength(1)
    wrapper.instance().addNewPin(0, 0)
    const PinnedColorListLen = wrapper.instance().state.pinnedColors.length
    expect(PinnedColorListLen).toEqual(2)
  })

  it('color pins could be moved by keyboard', () => {
    const wrapper = createColorsFromImage()
    wrapper.setState({
      isActivedPage: true,
      pinnedColors: pinnedColors
    })
    let canvasOffset = { x: 0, y: 0 }
    wrapper.instance().imageDataData = [0, 0, 0]
    window.sessionStorage.setItem('canvasOffset', JSON.stringify(canvasOffset))
    const move = jest.fn()
    wrapper.instance().handlePinMoveByKeyboard = move
    wrapper.instance().forceUpdate()
    wrapper.find(ColorsFromImagePin).props().handlePinMoveByKeyboard()
    expect(move).toHaveBeenCalled()
  })
})
