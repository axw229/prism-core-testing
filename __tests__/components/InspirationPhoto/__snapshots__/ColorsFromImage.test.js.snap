// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot match testing snapshot testing 1`] = `
<div
  className="scene__image__wrapper"
  onClick={[Function]}
  role="presentation"
>
  <canvas
    className="scene__image__wrapper__canvas"
    name="canvas"
  />
  <img
    alt=""
    className="scene__image__wrapper__image"
    onError={[Function]}
    onLoad={[Function]}
    src="src/images/inspirational-photos/green-thumb/GreenThumb1.jpg"
  />
</div>
`;
